pkg/cloudprovider/providers/aws/aws_metrics.go

```diff
@@ -53,8 +71,67 @@ func recordAWSThrottlesMetric(operation string) {
 	awsAPIThrottlesMetric.With(prometheus.Labels{"operation_name": operation}).Inc()
 }
 
-func registerMetrics() {
+func newVolumeStateCollector(cloud *Cloud) *volumeStateCollector {
+	return &volumeStateCollector{cloud}
+}
+
+type volumeStateCollector struct {
+	c *Cloud
+}
+
+var _ prometheus.Collector = &volumeStateCollector{}
+
+func (vsc *volumeStateCollector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- awsVolumesInStateTotal
+	ch <- awsVolumesInStateDurationMinutes
+}
+
+func (vsc *volumeStateCollector) Collect(ch chan<- prometheus.Metric) {
```

Comment:
This will be hard to cover via e2e. We should get some unit tests going may be and extract the function that creates metrics into separate function or something and check the return value.

Reply:
refactored and added tests